scalableminds/webknossos
# Allow to restrict floodfill to a bounding box

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- check out the new toggle in the tool bar when the floodfill tool is selected
![image](https://github.com/user-attachments/assets/f2ca8697-d531-4c05-8212-36a881f16d66)
- when disabled, the flood fill should behave almost as usual
  - one change is that if it terminates early, it should create a new bounding box only if in 3D mode
- when enabled, the flood fill operation will be constrained to the bbox that encloses the clicked position. if there is none, a toast should appear

### TODOs:
- [x] better icon
- [x] don't create new bbox in 3D case if the early termination was due to the custom bbox restriction
- [x] what about huge bboxes?
  - [x] blocked by icon https://scm.slack.com/archives/C5AKLAV0B/p1734449630499789?thread_ts=1733821465.594899&cid=C5AKLAV0B

### Issues:
- fixes #8259 

------
(Please delete unneeded items, merge only when none are left open)
